clang/lib/Tooling/DependencyScanning/DependencyScanningWorker.cpp

```diff
@@ -193,8 +210,18 @@ static void sanitizeDiagOpts(DiagnosticOptions &DiagOpts) {
   DiagOpts.ShowCarets = false;
   // Don't write out diagnostic file.
   DiagOpts.DiagnosticSerializationFile.clear();
-  // Don't emit warnings as errors (and all other warnings too).
-  DiagOpts.IgnoreWarnings = true;
+  // Don't emit warnings except for scanning specific warnings.
+  // TODO: It would be useful to add a more principled way to ignore all
+  //       warnings that come from source code. The issue is that we need to
+  //       ignore warnings that could be surpressed by
+  //       `#pragma clang diagnostic`, while still allowing some scanning
+  //       warnings for things we're not ready to turn into errors yet.
+  //       See `test/ClangScanDeps/diagnostic-pragmas.c` for an example.
+  llvm::erase_if(DiagOpts.Warnings, [](StringRef Warning) {
+    return llvm::StringSwitch<bool>(Warning)
+        .Cases("pch-vfs-diff", "error=pch-vfs-diff", false)
+        .Default(true);
```

Comment:
llvm/llvm-project#82294 has an extra `.StartsWith("no-error=", false)` in here. Was there a reason for not including it? In my testing it seems to be necessary to avoid a test failure in [`ClangScanDeps/diagnostics.c`](https://github.com/apple/llvm-project/blob/stable/20230725/clang/test/ClangScanDeps/diagnostics.c), which checks for `-no-error=` being ignored.

Reply:
@shahmishal: did something change about "please test llvm"? I am looking at the logs in https://ci.swift.org/view/Pull%20Requests/job/apple-llvm-project-pr-macos/ and only LLDB tests seem to run during those? I cannot find the LLVM or Clang tests (but the logs are 134MB, so I might be not looking in the right place).